pkts-streams/src/main/java/io/pkts/streams/impl/tcpFSM/TcpStreamFSM.java

```diff
@@ -0,0 +1,140 @@
+package io.pkts.streams.impl.tcpFSM;
+
+import io.hektor.fsm.Definition;
+import io.hektor.fsm.FSM;
+import io.hektor.fsm.builder.FSMBuilder;
+import io.hektor.fsm.builder.StateBuilder;
+import io.pkts.packet.TCPPacket;
+import io.pkts.streams.impl.TransportStreamId;
+
+import static io.pkts.streams.impl.tcpFSM.TcpStreamFSM.TcpState.*;
+
+public class TcpStreamFSM{
+
+    public enum TcpState{
+        INIT, HANDSHAKE, ESTABLISHED, FIN_WAIT_1, FIN_WAIT_2,
+        CLOSING_1_CLOSING_2, CLOSED_1_CLOSING_2, CLOSING_1_CLOSED_2, CLOSED
+    }
+
+    public final static Definition<TcpState, TcpStreamContext, TcpStreamData> definition;
+
+    static {
```

Comment:
A bit of a nit; but copy/paste the TCP state machine into the top here as it is much easier to understand what's going on.